packages/flutter/test/material/chip_test.dart

```diff
@@ -1145,4 +1147,73 @@ void main() {
     expect(materialBox, paints..path(color: customTheme.disabledColor));
     expect(labelStyle.style.color, equals(Colors.black.withAlpha(0xde)));
   });
+
+  testWidgets('Chip semantics', (WidgetTester tester) async {
```

Comment:
Can you add some more tests? Would be good to have one that correctly shows that the flag for selected is only present when onSelected is passed in. Similarly, we should ensure that SemanticsFlag.hasEnabledState is not present when the chip is not enable-able.